dvc/repo/get.py

```diff
@@ -20,7 +20,9 @@ def get(url, path, out=None, rev=None):
     dpath = os.path.dirname(os.path.abspath(out))
     tmp_dir = os.path.join(dpath, "." + str(shortuuid.uuid()))
     try:
-        with external_repo(cache_dir=tmp_dir, url=url, rev=rev) as repo:
+        with external_repo(
+            cache_dir=tmp_dir, url=url, rev=rev, no_state=True
+        ) as repo:
             # Try any links possible to avoid data duplication.
```

Comment:
The only change should be here:
```python
repo.state = ...
```

Reply:
Indeed, state db is not connected to until `__enter__`, so we could do that.